src/app/fdctl/run/tiles/fd_repair.c

```diff
@@ -0,0 +1,675 @@
+                   strcmp( topo->links[ tile->in_link_id[ NET_IN_IDX     ] ].name, "net_repair")     ||
+                   strcmp( topo->links[ tile->in_link_id[ CONTACT_IN_IDX ] ].name, "gossip_repair" ) ||
+                   strcmp( topo->links[ tile->in_link_id[ STAKE_IN_IDX ] ].name,   "stake_out" )     ||
+                   strcmp( topo->links[ tile->in_link_id[ STORE_IN_IDX ] ].name,   "store_repair" ) ) ) {
+    FD_LOG_ERR(( "Repair tile has none or unexpected input links %lu %s %s",
+                 tile->in_cnt, topo->links[ tile->in_link_id[ 0 ] ].name, topo->links[ tile->in_link_id[ 1 ] ].name ));
+  }
+
+  if( FD_UNLIKELY( tile->out_cnt != 1 ||
+                   strcmp( topo->links[ tile->out_link_id[ STORE_OUT_IDX ] ].name, "repair_store" ) ) ) {
+    FD_LOG_ERR(( "repair tile has none or unexpected output links %lu %s %s",
+                 tile->out_cnt, topo->links[ tile->out_link_id[ 0 ] ].name, topo->links[ tile->out_link_id[ 1 ] ].name ));
+  }
+
+  if( FD_UNLIKELY( tile->out_link_id_primary == ULONG_MAX ) )
+    FD_LOG_ERR(( "repair tile has no primary output link" ));
+
+  /* Scratch mem setup */
+
+  FD_SCRATCH_ALLOC_INIT( l, scratch );
+  fd_repair_tile_ctx_t * ctx = FD_SCRATCH_ALLOC_APPEND( l, alignof(fd_repair_tile_ctx_t), sizeof(fd_repair_tile_ctx_t) );
+  ctx->repair = FD_SCRATCH_ALLOC_APPEND( l, fd_repair_align(), fd_repair_footprint() );
+
+  void * smem = FD_SCRATCH_ALLOC_APPEND( l, fd_scratch_smem_align(), fd_scratch_smem_footprint( SMAX ) );
+  void * fmem = FD_SCRATCH_ALLOC_APPEND( l, fd_scratch_fmem_align(), fd_scratch_fmem_footprint( SDEPTH ) );
+
+  FD_TEST( ( !!smem ) & ( !!fmem ) );
+  fd_scratch_attach( smem, fmem, SMAX, SDEPTH );
+  
+  ulong scratch_top = FD_SCRATCH_ALLOC_FINI( l, 1UL );
+  if( FD_UNLIKELY( scratch_top > (ulong)scratch + scratch_footprint( tile ) ) )
+    FD_LOG_ERR(( "scratch overflow %lu %lu %lu", scratch_top - (ulong)scratch - scratch_footprint( tile ), scratch_top, (ulong)scratch + scratch_footprint( tile ) ));
+  
+  ctx->wksp = topo->workspaces[ topo->objs[ tile->tile_obj_id ].wksp_id ].wksp;
+
+  void * alloc_shmem = fd_wksp_alloc_laddr( ctx->wksp, fd_alloc_align(), fd_alloc_footprint(), 3UL );
+  if( FD_UNLIKELY( !alloc_shmem ) ) { 
+    FD_LOG_ERR( ( "fd_alloc too large for workspace" ) ); 
+  }
+  
+  ctx->repair_intake_listen_port = tile->repair.repair_intake_listen_port;
+  ctx->repair_serve_listen_port = tile->repair.repair_serve_listen_port;
+
+  FD_TEST( ctx->repair_intake_listen_port!=0 );
+  FD_TEST( ctx->repair_serve_listen_port!=0 );
+
+  fd_topo_link_t * netmux_link = &topo->links[ tile->in_link_id[ 0 ] ];
+
+  ctx->net_in    = topo->workspaces[ topo->objs[ netmux_link->dcache_obj_id ].wksp_id ].wksp;
+  ctx->chunk  = fd_disco_compact_chunk0( ctx->net_in );
+  ctx->wmark  = fd_disco_compact_wmark ( ctx->net_in, netmux_link->mtu );
+
+  // TODO: make configurable
+  FD_TEST( getrandom( ctx->identity_private_key, 32UL, 0 ) == 32UL );
```

Comment:
Is this supposed to use a random public key? Not the validator identity key?

Also this will fail. getrandom is not available in the sandbox for this tile.